test/middleware/AuthMiddleware.test.ts

```diff
@@ -0,0 +1,127 @@
+import 'reflect-metadata';
+import '../../src/controllers/index';
+import { expect, assert } from 'chai';
+import * as request from 'supertest';
+import { PENALTY_DETAILS_PAGE_URI, OTHER_REASON_DISCLAIMER_PAGE_URI, OTHER_REASON_PAGE_URI } from '../../src/utils/Paths';
+import { createApp, getDefaultConfig } from '../ApplicationFactory';
+import { Maybe, VerifiedSession } from 'ch-node-session-handler';
+import { createFakeSession } from '../utils/session/FakeSessionFactory';
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { generateSessionId, generateSignature } from 'ch-node-session-handler/lib/utils/CookieUtils';
+
+const config = getDefaultConfig();
+const id = generateSessionId();
+const sig = generateSignature(id, config.cookieSecret);
+
+const session = createFakeSession(
+    [{
+        [SessionKey.Id]: id
+    }, {
+        [SessionKey.ClientSig]: sig
+    }], config.cookieSecret, true);
+
+const authedApp = createApp(session);
+
+const protectedPages = [
+    PENALTY_DETAILS_PAGE_URI,
+    OTHER_REASON_DISCLAIMER_PAGE_URI,
+    OTHER_REASON_PAGE_URI
+];
+
+describe('Authentication Middleware', () => {
+
+
+    describe('Authed path', () => {
+
+        let penaltyDetails: Record<string, any>;
+
+        before('sign in server should be running', async () => {
+
+            penaltyDetails = {
+                penaltyReference: 'A12345678',
+                companyNumber: 'SC123123'
+            };
+
+            await request(createApp())
```

Comment:
I think there is no point having this precondition here. Application that is created and tested here is thrown away and not used in any of the tests executed afterwards.